src/libawkward/array/EmptyArray.cpp

```diff
@@ -377,6 +383,25 @@ namespace awkward {
 
   const std::string
   EmptyArray::validityerror(const std::string& path) const {
+    if (parameters_.size() != 0) {
+      bool result = std::any_of(valid_parameters.begin(), valid_parameters.end(),
+        [&](const std::string& i){
+          return (parameter_equals("__array__", i)) ? true : false;
+        });
+    if (result) {
+      // FIXME: further checks
+    }
+    result = std::any_of(non_valid_parameters.begin(), non_valid_parameters.end(),
+      [&](const std::string& i){
+        return (parameter_equals("__array__", i)) ? true : false;
+      });
+      if (result) {
+        return (std::string("at ") + path + std::string(" (") + classname()
+                + std::string("): __array__ can not be ")
+                + util::parameter_asstring(parameters_, "__array__")
+                + FILENAME(__LINE__));
+      }
+    }
```

Comment:
Personally, I would just write a `for` loop, or even a single `if` statement with explicit checks. The complexity cost of introducing lambda functions and `std::any_of` for something this simple doesn't seem worth it.

If we want to allow for growth in the set of allowed and not-allowed parameters, we can just centralize this check using an internal function call, maybe a method on `Content`. That would also be more DRY than listing the set of allowed/not-allowed parameters in every node's .cpp file. When we add a new recognized parameter with rules about which nodes it can be used on, do we really want to add it to the not-allowed list of every .cpp file except one? That would be a big maintenance burden.